app/models/course_holes_stat_log.rb

```diff
@@ -0,0 +1,19 @@
+class CourseHolesStatLog
+  attr_reader :hole_stat_logs
+
+  def initialize(course: course, user: user)
+    @hole_stat_logs = course.holes.map { |hole| HoleStatLog.new(hole: hole, user: user) }
+    assign_hole_rankings
+  end
+
+  private
+
+  def assign_hole_rankings
+    sorted_average_shooting = hole_stat_logs.map(&:average_shooting).compact.sort.uniq
+    hole_stat_logs.each do |log|
+      if ranking = sorted_average_shooting.index(log.average_shooting)
```

Comment:
Is there a version of this that adds the +1 for you? Ala `3.times do` vs `1.upto 3 do`